### PR TITLE
Update to Cairo 0.9.1, starknet.py 0.4.1

### PR DESCRIPTION
**Pull request by MaksymilianDemitraszek**

### CODE REVIEW

**Review by Radinyn — APPROVED** [review]

File: pyproject.toml [hunk 1]
```diff
@@ -10,19 +10,19 @@ version = "0.2.6"
...
 pytest-datadir = "^1.3.1"
 python = ">=3.7.12, <3.10"
 requests = "^2.27.1"
-"starknet.py" = "^0.3.0-alpha.0"
 tomli = "<2.0.0"
 tomli-w = "^1.0.0"
 tqdm = "^4.64.0"
 typing-extensions = "^4.0.1"
+"starknet.py" = "^0.4.1-alpha.0"
+cairo-lang = "^0.9.1"
```
Comment (mkaput) [inline pyproject.toml:25]: Could you please keep this sorted?

File: tests/integration/constructor_in_tested_file/error_from_deploy_syscall_test.cairo [hunk 2]
```diff
@@ -11,7 +11,7 @@ func test_deploy_declared_contract{syscall_ptr : felt*, range_check_ptr}():
     %{ ids.class_hash = declare("./tests/integration/constructor_in_tested_file/basic_contract.cairo").class_hash %}
 
     let (local calldata : felt*) = alloc()
-    let (contract_address) = deploy(class_hash, 42, 0, calldata)
+    let (contract_address) = deploy(class_hash, 42, 0, calldata, 0)
```
Comment (mkaput) [inline tests/integration/constructor_in_tested_file/error_from_deploy_syscall_test.cairo:14]: Seems like docs for this cheat code are not changed?

**Review by mkaput — CHANGES_REQUESTED** [review]

File: tests/integration/constructor_in_tested_file/error_from_deploy_syscall_test.cairo (hunk 2 above)
Comment (kasperski95) [reply tests/integration/constructor_in_tested_file/error_from_deploy_syscall_test.cairo:14]: It's a syscall, not a cheatcode.

Comment (mkaput) [reply tests/integration/constructor_in_tested_file/error_from_deploy_syscall_test.cairo:14]: ah ok

**Review by kasperski95 — COMMENTED** [review]
We need to update `deploy_command` in this or another PR.

**Comment by MaksymilianDemitraszek** [comment]
@kasperski95 https://github.com/software-mansion/protostar/issues/515

**Review by mkaput — APPROVED** [review]

**Pull request merged**